### PR TITLE
512 training

### PR DESCRIPTION
**Pull request by williamberman**

### CODE REVIEW

File: muse/modeling_transformer.py [hunk 1]
```diff
@@ -1589,7 +1590,9 @@ def __init__(
             input_channels = output_channels
             output_channels = block_out_channels[i]
 
-            if use_vannilla_resblock:
+            if force_down_up_sample:
+                add_downsample = True
```
Comment (patil-suraj) [inline muse/modeling_transformer.py:1594]: so here we are forcing up/down sampling in existing blocks instead of adding a new one ?

File: training/train_muse.py [hunk 2]
```diff
@@ -159,8 +160,25 @@ def mask_or_random_replace_tokens(image_tokens, mask_id, config, mask_schedule,
...
+        batch_randperm = torch.rand(batch_size, seq_len, device=image_tokens.device).argsort(dim=-1)
+        mask = batch_randperm < num_token_masked.unsqueeze(-1)
+    else:
+        # TODO - would be nice to vectorize
+        mask = torch.zeros((batch_size, seq_len), device=image_tokens.device)
+        for batch_idx, num_token_masked_ in enumerate(num_token_masked):
+            num_token_masked_ = int(num_token_masked_.item())
+            start_idx = random.randint(0, seq_len - num_token_masked_)
+            mask[batch_idx, start_idx : start_idx + num_token_masked_] = 1
+        mask = mask.to(torch.bool)
```
Comment (patil-suraj) [inline training/train_muse.py:181]: here maybe it would be better to treat this as 2D and the mask contineous region. That should correspond to actual image masking. wdyt ?

**Review by patil-suraj — COMMENTED** [review]
Looks great! Just left two que

File: muse/modeling_transformer.py (hunk 1 above)
Comment (williamberman) [reply muse/modeling_transformer.py:1594]: yep, just less additional weights

File: training/train_muse.py (hunk 2 above)
Comment (williamberman) [reply training/train_muse.py:181]: yes, agreed
